Networking Working Group

alexandruradovici opened the pull request:
### Pull Request Overview

This pull request defines the charter of the networking group.

[Rendered](https://github.com/OxidosAutomotive/tock/tree/network_working_group/doc/wg/network)


### Testing Strategy

N/A


### TODO or Help Wanted

Feedback is welcome, @brghena I might have missed something from our discussion about the attributions of the group.


### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [ ] Ran `make prepush`.

lschuermann commented on doc/wg/network/README.md:
```diff
@@ -0,0 +1,43 @@
+the group: the exact process is not yet determined and may organically
+evolve as the group gains momentum.
+
+The group has a weekly teleconference call. All working group members
```
```suggestion
The group has a biweekly teleconference call. All working group members
```

Assuming we stick to what we talked about at TockWorld 6. Not strictly opposed to weekly though.

brghena replied:
"biweekly" is actually ambiguous in English and means either twice a week or once every two weeks. Both meanings are common which really just means that the word is meaningless.

```suggestion
The group has a teleconference call once every two weeks. All working group members
```

lschuermann commented on doc/wg/network/README.md:
```diff
@@ -0,0 +1,43 @@
+## Members
+
+- Alexandru Radovici, OxidOS Automotive
+- Brandon Ghena, Northwestern University
```
```suggestion
- Branden Ghena, Northwestern University
```

lschuermann reviewed (approved):
Looks good to me.

brghena commented on doc/wg/network/README.md:
```diff
@@ -0,0 +1,43 @@
+- Alexandru Radovici, OxidOS Automotive
+- Brandon Ghena, Northwestern University
+- Leon Schuermann, Princeton University
+- Tyler Potyondy, UCSD
```
@tyler-potyondy I wanted to confirm that you're interested in being part of the working group (you can always leave in a couple months if it no longer makes sense to join calls).

brghena commented on doc/wg/network/README.md:
```diff
@@ -0,0 +1,43 @@
+
+The goals of the Network Tock Working Group (NWG) are to:
+
+- design a set of interfaces (traits) for networking
+- design the user land interfaces for networking
+- define how buffer management should be implemented
```
```suggestion
- design a set of interfaces (traits) for networking
- design the user land interfaces for networking
- define how buffer management should be implemented
- determine other kernel infrastructure useful for networking
```

brghena commented on doc/wg/network/README.md:
```diff
@@ -0,0 +1,43 @@
+posted within a week of a call. This delay is to give participants an
+opportunity to correct any errors or better explain points that came up.
+They are intended to be a communication mechanism of the group, its
+discussions, the technical issues, and decisions, not a literal
+transcription of what is said.
```
```suggestion
discussions, the technical issues, and decisions, not a literal
transcription of what is said.

## Code Purview

At this time, the network working group does not have responsibility for code in any particular directories of Tock. That may change as work progresses, with the possibility of a subdirectory within `capsules/` being created under the purview of this working group.
```

brghena commented:
A couple of small proposed changes, but this looks great.

bradjc commented:
Who is chair?

brghena commented:
Alexandru, unless he doesn't want to, in which case I'll do it. (should be added to the writeup though)

alexandruradovici commented:
> Alexandru, unless he doesn't want to, in which case I'll do it. (should be added to the writeup though)

Happy to do it with help from @brghena

lschuermann commented on doc/wg/network/README.md:
```diff
@@ -0,0 +1,48 @@
+
+## Membership and Communication
+
+The networking working group membership is a subset of the people who have
+commit (pull request merge) permissions on the Tock repository. It is
```
Is this true in practice (I think the answer is now) and do we want this to be the case? I don't see any hard requirement for this.

brghena replied:
```suggestion
The networking working group membership is open to Tock developers interested
in the design of network interfaces. It is
```
What about this?

lschuermann replied:
I don‘t see this addressed in the current version.

ppannuto commented on doc/wg/network/README.md:
```diff
@@ -0,0 +1,48 @@
+help contribute to particular topics or on-going discussions. The
+working group chair decides who beyond the working group members may
+participate in the call.
+
```
```suggestion
Those looking to engage with the working group are encouraged to join the `#networking-working-group` channel on the [Tock slack](https://github.com/tock/tock/#keep-up-to-date).
```

lschuermann replied:
Same here, not yet addressed.

bradjc commented on doc/wg/network/README.md:
```diff
@@ -0,0 +1,48 @@
+working group chair decides who beyond the working group members may
+participate in the call.
+
+The working group publishes detailed notes of its calls. These will be
```
```suggestion

The working group publishes detailed notes of its calls. These will be
```

brghena replied:
I don't understand what the suggestion is here. The "suggested change" box has the - and + exactly identical. Unless it's that you added a newline??

alexandruradovici commented:
As discussed with @brghena, I deleted Ionut Cirstea from the list and added:
- Cristian Rusu from the University of Bucharest, he is doing research in OS and using Tock
- Felix Mada from OxidOS Automotive, he is responsible for implementing the CAN driver

My contribution here will be from the Politehnica University

lschuermann reviewed (changes requested):
If I‘m not mistaken (and GitHub doesn‘t trick me), two of the comments are not yet addressed in the current version. Especially the commit access requirement is, I think, something we don‘t want.

alevy commented:
@alexandruradovici ping on the final two small changes?

alexandruradovici commented:
I think this should be ready.

bradjc commented on doc/wg/network/README.md:
```diff
@@ -0,0 +1,49 @@
+=================================
+
+- Working Group Charter
+- Adopted 7/28/2023
```
```suggestion
- Adopted 8/11/2023
```
